### PR TITLE
DOCKER-322 Speed up build

### PR DESCRIPTION
**Pull request by vierbergenlars**
1. Update docker-alfresco plugin to 5.0.5
2. Load init-jar through configurations
3. Consolidate copy commands
4. Move `ARG` to the end of Dockerfile, so build cache can be used more often.
5. Custom image tags for non-release builds (to avoid conflicts when running builds for multiple branches in Jenkins)
6. Run parallel stages with build and integration test for each version instead of parallel builds and integration tests across versions
7. Keep same BUILD_DATE across all task executions to make use of build cache

### CODE REVIEW

**Comment by vierbergenlars** [comment]
Reduces Jenkins buildtime from ~1h10m to ~40m

File: 2repository/build.gradle [hunk 1]
```diff
@@ -38,48 +39,33 @@ subprojects {
...
-        createDockerFile.dependsOn(parentImage)
-
-        if (project.hasProperty('testsChangedPorts') && project.testsChangedPorts)
-            changedPortsComposeUp.dependsOn(buildDockerImage)
-        else
-            changedPortsComposeUp.enabled = false
+    createDockerFile {
+        smartCopy(configurations.sharedLib, '${CATALINA_HOME}/shared/lib/')
+        if (!configurations.sharedBin.isEmpty())
+            smartCopy(tarTree(resources.gzip(configurations.sharedBin.singleFile)), '/opt/alfresco')
```
Comment (tgeens) [inline 2repository/build.gradle:53]: sharedBin is always a single file ?

File: 3legacy/5.1/enterprise-5.1.5/overload.gradle [hunk 2]
```diff
@@ -11,5 +11,4 @@ ext {
 
 dependencies {
     baseShareWar 'org.alfresco:share:5.1.5@war'
-    sharedLib(group: 'org.postgresql', name: 'postgresql', version: '9.4.1212')
```
Comment (tgeens) [inline 3legacy/5.1/enterprise-5.1.5/overload.gradle:14]: why don't we need the postgres driver anymore ?

File: 2repository/build.gradle (hunk 1 above)
Comment (vierbergenlars) [reply 2repository/build.gradle:53]: It was already like that before. It is a `.tar.gz` file that needs to be unzipped before being extracted.
It is the `alfresco-pdf-renderer`, so perhaps I can rename the configuration.

https://github.com/xenit-eu/docker-alfresco/blob/9a03886f2cf50b7fc9f2c2c2505f891179f1338c/3legacy/5.2/enterprise-5.2.5/overload.gradle#L14

File: 3legacy/5.1/enterprise-5.1.5/overload.gradle (hunk 2 above)
Comment (vierbergenlars) [reply 3legacy/5.1/enterprise-5.1.5/overload.gradle:14]: The postgres driver is already in there.
Legacy images build on top of repository images.
The postgres driver is already added in the repository image that is underneath.
The only thing that needs to be added in the legacy subproject is Share.

I checked the built image for presence of the driver.

```
$ docker run hub.xenit.eu/lvierbergen/alfresco-enterprise:5.1.5-build-DOCKER-322-32 stat /usr/local/tomcat/shared/lib/postgresql-9.4.1212.jar
> Executing /docker-entrypoint.d/80-java-opts.sh
Result: JAVA_OPTS=
> Executing /docker-entrypoint.d/85-init-tomcat.sh
Tomcat init start
Tomcat init done
> Executing /docker-entrypoint.d/89-check-validity.sh
> Executing /docker-entrypoint.d/90-init-alfresco.sh
Alfresco init start
Alfresco init done
> Executing /docker-entrypoint.d/91-init-share-customized.sh
Share init start
Share init done
> Starting CMD: stat /usr/local/tomcat/shared/lib/postgresql-9.4.1212.jar
> exec stat /usr/local/tomcat/shared/lib/postgresql-9.4.1212.jar
  File: /usr/local/tomcat/shared/lib/postgresql-9.4.1212.jar
  Size: 680445    	Blocks: 1336       IO Block: 4096   regular file
Device: 77h/119d	Inode: 19690087    Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2020-05-07 12:08:22.000000000 +0000
Modify: 2020-05-07 12:08:22.000000000 +0000
Change: 2020-05-07 13:50:07.012357816 +0000
 Birth: -
```

**Pull request merged**